core/core-auto-completion.el

```diff
@@ -34,10 +34,11 @@ The initialization function is hooked to `MODE-hook'."
               'company)
          (set (make-variable-buffer-local 'company-backends)
               ,backend-list))
-       (when auto-completion-enable-snippets-in-popup
-         (setq ,backend-list (mapcar 'spacemacs//show-snippets-in-company
-                                     ,backend-list)))
        (add-hook ',mode-hook ',func t)
+       (add-hook ',mode-hook (lambda ()
+                               (when auto-completion-enable-snippets-in-popup
+                                 (setq ,backend-list (mapcar 'spacemacs//show-snippets-in-company
+                                                             ,backend-list)))))
```

Comment:
Did you try by putting your code to the end or the beginning of `,func` ?
If you put it at the beginning it should work because the only thing that your additional lambda does is to execute the code before `,func`.